attestation/system-packages/system-packages.go

```diff
@@ -0,0 +1,161 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package systempackages
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/in-toto/go-witness/attestation"
+	"github.com/invopop/jsonschema"
+)
+
+const (
+	Name    = "system-packages"
+	Type    = "https://witness.dev/attestations/system-packages/v0.1"
+	RunType = attestation.PreMaterialRunType
+)
+
+func init() {
+	attestation.RegisterAttestation(Name, Type, RunType, func() attestation.Attestor {
+		return NewSystemPackagesAttestor()
+	})
+}
+
+type Attestor struct {
+	OS           string    `json:"os"`
+	Distribution string    `json:"distribution"`
+	Version      string    `json:"version"`
+	Packages     []Package `json:"packages"`
+	backend      Backend
+}
+
+type Package struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+type Backend interface {
+	DetermineOSInfo() (string, string, string, error)
+	GatherPackages() ([]Package, error)
+	SetExecCommand(cmd func(name string, arg ...string) *exec.Cmd)
+}
+
+func NewSystemPackagesAttestor() *Attestor {
+	osReleaseFile := "/etc/os-release"
+	_, distribution, _, err := determineDistribution(osReleaseFile)
+	fmt.Println("discovered distribution:", distribution)
+	if err != nil {
+		// Default to Debian-based system if we can't determine the distribution
+		return &Attestor{
+			backend: NewDebianBackend(osReleaseFile),
+		}
+	}
+
+	switch distribution {
+	case "fedora", "rhel", "centos", "rocky", "alma", "oracle", "suse", "opensuse", "amazon":
+		return &Attestor{
+			backend: NewRPMBackend(osReleaseFile),
+		}
+	case "debian", "ubuntu":
+		return &Attestor{
+			backend: NewDebianBackend(osReleaseFile),
+		}
+	default:
+		// Use Debian backend for any other unrecognized distributions
+		return &Attestor{
+			backend: NewDebianBackend(osReleaseFile),
+		}
+	}
+}
+
+func determineDistribution(osReleaseFile string) (string, string, string, error) {
+	file, err := os.Open(osReleaseFile)
+	if err != nil {
+		return "", "", "", err
+	}
+	defer file.Close()
+
+	var distribution, version string
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) != 2 {
+			continue
+		}
+		key := strings.TrimSpace(parts[0])
+		value := strings.Trim(strings.TrimSpace(parts[1]), "\"")
+
+		switch key {
+		case "ID":
+			distribution = value
+		case "VERSION_ID":
+			version = value
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return "", "", "", err
+	}
+
+	return "linux", distribution, version, nil
+}
+
+// Attest implements attestation.Attestor.
+func (a *Attestor) Attest(ctx *attestation.AttestationContext) error {
+	os, dist, version, err := a.backend.DetermineOSInfo()
+	fmt.Println(os)
```

Comment:
Can we use the logger here?  I have a console logger I am using to test this branch.  https://github.com/in-toto/go-witness/pull/408